specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/ReplicationLinks.json

```diff
@@ -109,6 +109,64 @@
+          },
+          "202": {
+            "description": "Update operation is in progress."
```

Comment:
PUT should return 201 and 200, not 202, as described here: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creating-or-updating-resources-asynchronously).

Reply:
It is a special case of PUT where the functionality is exactly same as PATCH request. It only allows updation of the link type. It does not support the creation of the link.  The PUT controller has been added along with PATCH to make it consistent with the ARM deployments templates. 
I see other PUT requests returning 202 as well: https://github.com/lobashh/azure-rest-api-specs/blob/cbb06631004dca161ae1ed2d02bb7dea953c4a0f/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/examples/FailoverGroupCreateOrUpdate.json

Reply:
This is not really allowed anymore. THe 202 scheme has been retired a long back. Also can you clearify what do you mean by special type of PUT operation ? What would happen in this case if the resource does not exist and you do a PUT on this resource ?

Reply:
PUT here would only support in place Update of the link type (just like PATCH) and it would not allow the creation of the link (would throw 404 with error message) and would not support updating any other parameter. 

Keeping in sync with the implementation of many other SQL APIs and the other APIs of Geo DR (to which Replication Link entity belongs), we chose to proceed with 202. Also, it is mentioned in the document that it can be used e.g https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#1321-put 
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers 
As it is simple update operation, changing it at this point will cause great implementation overhead.

Reply:
How is the resource created without a PUT operation ?

Reply:
The replication link is created as part of creation of geo secondary replica. There is no PUT for creation of link

Reply:
For new PUT operation, please always use 200 and 201. it's a required check now.
https://github.com/Azure/azure-openapi-validator/blob/main/docs/put-response-codes.md